lib/ascii_binder/engine.rb

```diff
@@ -0,0 +1,872 @@
+            git_checkout(local_branch)
+          else
+            next
+          end
+        end
+
+        # Note the image files checked in to this branch.
+        branch_image_files = Find.find(source_dir).select{ |path| not path.nil? and (path =~ /.*\.png$/ or path =~ /.*\.png\.cache$/) }
+
+        first_branch = single_page.nil?
+
+        if local_branch =~ /^\(detached from .*\)/
+          local_branch = 'detached'
+        end
+
+        # The branch_orphan_files list starts with the set of all
+        # .adoc files found in the repo, and will be whittled
+        # down from there.
+        branch_orphan_files = find_topic_files
+        branch_build_config = build_config
+        remove_found_config_files(local_branch,branch_build_config,branch_orphan_files)
+
+        if branch_orphan_files.length > 0 and single_page.nil?
+          nl_warning "Branch '#{local_branch}' includes the following .adoc files that are not referenced in the #{build_config_file} file:\n" + branch_orphan_files.map{ |file| "- #{file}" }.join("\n")
+        end
+
+        # Run all distros.
+        distro_map.distros.each do |distro|
+          if not build_distro == ''
+            # Only building a single distro; build for all indicated branches, skip the others.
+            if not build_distro == distro.id
+              next
+            end
+          else
+            current_distro_branches = distro_map.distro_branches(distro.id)
+
+            # In publish mode we only build "valid" distro-branch combos from the distro map
+            if branch_group.to_s.start_with?("publish") and not current_distro_branches.include?(local_branch)
+              next
+            end
+
+            # In "build all" mode we build every distro on the working branch plus the publish distro-branch combos
+            if branch_group == :all and not local_branch == working_branch and not current_distro_branches.include?(local_branch)
+              next
+            end
+          end
+
+          site_name = distro.site.name
+
+          branch_config = AsciiBinder::DistroBranch.new('',{ "name" => "Branch Build", "dir" => local_branch },distro.name,distro.author)
+          dev_branch    = true
+          if distro.branch_ids.include?(local_branch)
+            branch_config = distro.branch(local_branch)
+            dev_branch    = false
+          end
+
+          if first_branch
+            puts "\nBuilding #{distro.name} for branch '#{local_branch}'"
+            first_branch = false
+          end
+
+          # Create the target dir
+          branch_path           = File.join(preview_dir,distro.id,branch_config.dir)
+          branch_stylesheet_dir = File.join(branch_path,STYLESHEET_DIRNAME)
+          branch_javascript_dir = File.join(branch_path,JAVASCRIPT_DIRNAME)
+          branch_image_dir      = File.join(branch_path,IMAGE_DIRNAME)
+
+          # Copy files into the preview area.
+          [[stylesheet_dir, '*css', branch_stylesheet_dir],
```

Comment:
My folders are in the root but it would be better to make them configurable. I "hide" everythin that I don't want parsed in folder with the _name denomination. This includes a PDF template and some generated PDF files. I want one of those copied to the packages files and one not. 

I like the idea of an include mapping file.